### PR TITLE
feat: new resource `spacelift_gitlab_integration`

### PR DESCRIPTION
**Pull request by tmeckel**
## Description of the change

This PR contains the new resource `spacelift_gitlab_integration` to create an GitLab integration for a Spacelift space.

> [!IMPORTANT]
> The test `TestGitLabIntegrationClearLabels` currently fails, because it's impossible to clear assigned labels with a GraphQL mutation. Using `nil` or `[]` (empty array) makes no difference.

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (non-breaking change that adds documentation)

## Related issues

None

### Development

- [ ] Lint rules pass locally
- [x] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development
- [x] Examples for new resources and data sources have been added
- [x] Default values have been documented in the description (e.g., "Dummy: (Boolean) Blah blah. Defaults to `false`.)
- [ ] If the action fails that checks the documentation: Run `go generate` to make sure the docs are up to date

### Code review

- [ ] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] Pull Request is no longer marked as "draft"
- [ ] Reviewers have been assigned
- [ ] Changes have been reviewed by at least one other engineer

### CODE REVIEW

**Comment by truszkowski** [comment]
@tmeckel
Thanks for the contribution❤️ 
We'll discuss it with the team and create a new PR with Your changes (secrets don't work with forks).

**Comment by peterdeme** [comment]
@tmeckel this is a bug in our backend, I opened a PR for the fix. It should be deployed within 2-3 working days.

File: .envrc
Comment (peterdeme) [inline .envrc:1]: this file should be in gitignore, no?

Comment (tmeckel) [reply .envrc:1]: @peterdeme  Nope the idea is that [`direnv`](https://direnv.net/) loads an `.env` file (i present) which should be created out of the `[test.env.tmpl](https://github.com/spacelift-io/terraform-provider-spacelift/blob/main/test.env.tmpl)` file. I know it's more about convenience than a feature.

**Comment by truszkowski** [comment]
I've added cloned PR -> https://github.com/spacelift-io/terraform-provider-spacelift/pull/557
We'll merge it soon @tmeckel.

**Comment by tmeckel** [comment]
Closing because #557 has been merged.

**Pull request closed without merging**